sql/builder/builder.go

```diff
@@ -0,0 +1,82 @@
+package builder
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/spacemeshos/go-spacemesh/sql"
+)
+
+type token string
+
+const (
+	Eq    token = "="
+	NotEq token = "!="
+	Gt    token = ">"
+	Gte   token = ">="
+	Lt    token = "<"
+	Lte   token = "<="
+	And   token = "and"
+	Where token = "where"
+)
+
+type field string
+
+const (
+	Epoch    field = "epoch"
+	Smesher  field = "pubkey"
+	Coinbase field = "coinbase"
+	Id       field = "id"
+	Offset   field = "offset"
+	Limit    field = "limit"
+	OrderBy  field = "order by"
+)
+
+type Op struct {
+	Field field
+	Token token
+	// Value will be type casted to one the expected types.
+	// Operation will panic if it doesn't match any of expected.
+	Value any
+}
+
+type Operations struct {
+	Filter []Op
+	Other  []Op
+}
+
+func FilterFrom(operations Operations) string {
+	var queryBuilder strings.Builder
+
+	for i, op := range operations.Filter {
+		if i == 0 {
+			queryBuilder.WriteString(" " + string(Where))
+		}
+		if i != 0 {
+			queryBuilder.WriteString(" " + string(And))
+		}
+		queryBuilder.WriteString(" " + string(op.Field) + " " + string(op.Token) + " ?" + strconv.Itoa(i+1))
```

Comment:
I think that both `And` and `Where` are only used here and are not meant to be used outside. How about just inlining them?
```suggestion
		if i == 0 {
			queryBuilder.WriteString(" where")
		} else {
			queryBuilder.WriteString(" and")
		}
		queryBuilder.WriteString(" " + string(op.Field) + " " + string(op.Token) + " ?" + strconv.Itoa(i+1))
```